dvc/repo/fetch.py

```diff
@@ -74,8 +74,12 @@ def _fetch_external(self, repo_url, repo_rev, files, jobs):
     failed = 0
     try:
         with external_repo(repo_url, repo_rev) as repo:
-            repo.cache.local.cache_dir = self.cache.local.cache_dir
+            if not hasattr(repo, "cache"):
+                return _fetch_external_git(
+                    self.cache.local, repo.root_dir, files
+                )
 
```

Comment:
For the record: discussed with @skshetry that we might be missing support git-files imported from dvc-repo.